### PR TITLE
Identity provider in v1.0

### PR DESCRIPTION
**Pull request by namkedia**

### CODE REVIEW

File: api-reference/v1.0/resources/openidconnectidentityprovider.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+|responseType|String|The response type describes the type of information sent back in the initial call to the authorization_endpoint of the custom identity provider. Possible values: `code` , `id_token` , `token`.  Required.|
+|scope|String|Scope defines the information and permissions you are looking to gather from your custom identity provider. OpenID Connect requests must contain the openid scope value in order to receive the ID token from the identity provider. Without the ID token, users are not able to sign in to Azure AD B2C using the custom identity provider. Other scopes can be appended, separated by a space. For more details about the scope limitations see [RFC6749 Section 3.3](https://tools.ietf.org/html/rfc6749#section-3.3). Required.|
+
+### responseMode value
+|Value|Description|
+:--------|:----------|
+|form_post|This response mode is recommended for best security. The response is transmitted via the HTTP POST method, with the code or token being encoded in the body using the application/x-www-form-urlencoded format.|
+|query|The code or token is returned as a query parameter.|
+
+### responseType value
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/openidconnectidentityprovider.md:51]:
```suggestion
### responseType values
```

File: api-reference/v1.0/resources/openidconnectidentityprovider.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+|responseMode|String|The response mode defines the method used to send data back from the custom identity provider to Azure AD B2C. Possible values: `form_post`, `query`. Required.|
+|responseType|String|The response type describes the type of information sent back in the initial call to the authorization_endpoint of the custom identity provider. Possible values: `code` , `id_token` , `token`.  Required.|
+|scope|String|Scope defines the information and permissions you are looking to gather from your custom identity provider. OpenID Connect requests must contain the openid scope value in order to receive the ID token from the identity provider. Without the ID token, users are not able to sign in to Azure AD B2C using the custom identity provider. Other scopes can be appended, separated by a space. For more details about the scope limitations see [RFC6749 Section 3.3](https://tools.ietf.org/html/rfc6749#section-3.3). Required.|
+
+### responseMode value
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/openidconnectidentityprovider.md:45]:
```suggestion
### responseMode values
```

File: api-reference/v1.0/resources/openidconnectidentityprovider.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+:--------|:----------|
+|form_post|This response mode is recommended for best security. The response is transmitted via the HTTP POST method, with the code or token being encoded in the body using the application/x-www-form-urlencoded format.|
+|query|The code or token is returned as a query parameter.|
+
+### responseType value
+|Value|Description|
+:--------|:----------|
+|code|As per the authorization code flow, a code will be returned back to Azure AD B2C. Azure AD B2C proceeds to call the token_endpoint to exchange the code for the token.|
+|id_token|An ID token is returned back to Azure AD B2C from the custom identity provider.|
+|token|An access token is returned back to Azure AD B2C from the custom identity provider. (This value is not supported by Azure AD B2C at the moment)|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/openidconnectidentityprovider.md:56]: These intended/apply to the previous comment? Just align with [Example 2 in the Enumerations guidance](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/103158/Enumerations?anchor=second-option%3A-in-parent-resource-topic).

File: api-reference/v1.0/resources/socialidentityprovider.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+|[Get](../api/identityproviderbase-get.md) |socialidentityprovider |Retrieve properties of a social identity provider.|
+|[Update](../api/identityproviderbase-update.md)|None|Update a social identity provider.|
+|[Delete](../api/identityproviderbase-delete.md)|None|Delete a social  identity provider.|
+|[List available provider types](../api/identityproviderbase-list-availableprovidertypes.md)|String collection|Retrieve all available identity provider types available in the tenant.|
+
+## Properties
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|clientId|String|The client identifier for the application obtained when registering the application with the identity provider. Required.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/socialidentityprovider.md:40]:
```suggestion
|clientId|String|The identifier for the client application obtained when registering the application with the identity provider. Required.|
```

File: changelog/Microsoft.DirectoryServices.json [hunk 5]
```diff
@@ -7383,6 +7383,152 @@
...
       "SubArea": ""
     },
+    {
+      "ChangeList": [
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Resource",
+          "ChangedApiName": "identityProviderBase",
+          "ChangeType": "Addition",
+          "Description": "To support configuring an identity provider in an Azure AD and Azure AD B2C tenant, introduced new resource type [identityProviderBase](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0) entity.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7393]: Adding a little info of what the API does sounds good. I recommend the following format

_Added the [identityProviderBase](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0) resource to support configuring an identity provider in an Azure AD and Azure AD B2C tenant._]

Recommendation applies to the other entries

File: changelog/Microsoft.DirectoryServices.json [hunk 6]
```diff
@@ -7383,6 +7383,152 @@
...
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProviderBase",
+          "ChangeType": "Addition",
+          "Description": "To support getting an identity provider configuration in an Azure AD and Azure AD B2C tenant, introduced [get](https://docs.microsoft.com/en-us/graph/api/identityproviderbase-get?view=graph-rest-1.0) operation.",
+          "Target": "identityProviderBase"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7445]: Previous comment for resources also applies to methods.

File: changelog/Microsoft.DirectoryServices.json [hunk 7]
```diff
@@ -7383,6 +7383,152 @@
...
+          "ChangeType": "Addition",
+          "Description": "To support listing available provider types of identity provider in an Azure AD and Azure AD B2C tenant, introduced [availableProviderType](https://docs.microsoft.com/en-us/graph/api/identityprovider-list-availableprovidertypes?view=graph-rest-1.0) operation.",
+          "Target": "identityProviderBase"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Resource",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
+          "Description": "Deprecated the [identityProvider](https://docs.microsoft.com/en-us/graph/api/resources/identityprovider?view=graph-rest-1.0) entity to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7473]: - Suggestion applies to other entries
- If all its methods are deprecated, just say "... and related operations and methods" instead and remove the corresponding entries

```suggestion
          "Description": "Deprecated the [identityProvider](https://docs.microsoft.com/en-us/graph/api/resources/identityprovider?view=graph-rest-1.0) resource to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 8]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated [delete](https://docs.microsoft.com/en-us/graph/api/identityprovider-delete?view=graph-rest-1.0) operations to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        }
+      ],
+      "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+      "Cloud": "prd",
+      "Version": "v1.0",
+      "CreatedDateTime": "2021-03-30T00:00:00",
+      "WorkloadArea": "Identity and sign-in",
+      "SubArea": ""
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7530]:
```suggestion
      "CreatedDateTime": "2021-03-30T00:00:00.000Z",
      "WorkloadArea": "Identity and access",
      "SubArea": "Identity and sign-in"
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]

File: changelog/Microsoft.DirectoryServices.json [hunk 9]
```diff
@@ -7383,6 +7383,152 @@
...
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
+          "Description": "Deprecated [availableProviderType](https://docs.microsoft.com/en-us/graph/api/identityprovider-list-availableprovidertypes?view=graph-rest-1.0) operations to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7512]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 10]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated [update](https://docs.microsoft.com/en-us/graph/api/identityprovider-update?view=graph-rest-1.0) operations to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7520]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 11]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated [get](https://docs.microsoft.com/en-us/graph/api/identityprovider-get?view=graph-rest-1.0) operations to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7504]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 12]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated  [list](https://docs.microsoft.com/en-us/graph/api/identityprovider-list?view=graph-rest-1.0) operations to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7496]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 13]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated [post](https://docs.microsoft.com/en-us/graph/api/identityprovider-post-identityproviders?view=graph-rest-1.0) operation to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7488]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 14]
```diff
@@ -7383,6 +7383,152 @@
...
+          "Description": "Deprecated the [identityProvider](https://docs.microsoft.com/en-us/graph/api/resources/identityprovider?view=graph-rest-1.0) entity to replace it with new [identity provider API](https://docs.microsoft.com/en-us/graph/api/resources/identityproviderbase?view=graph-rest-1.0).",
+          "Target": "identityProvider"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Method",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7480]:
```suggestion
          "ChangeType": "Deprecation",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 15]
```diff
@@ -7383,6 +7383,152 @@
...
+          "ChangedApiName": "identityProviderBase",
+          "ChangeType": "Addition",
+          "Description": "To support listing available provider types of identity provider in an Azure AD and Azure AD B2C tenant, introduced [availableProviderType](https://docs.microsoft.com/en-us/graph/api/identityprovider-list-availableprovidertypes?view=graph-rest-1.0) operation.",
+          "Target": "identityProviderBase"
+        },
+        {
+          "Id": "224526a5-09a4-4895-9b8b-40e8984a4111",
+          "ApiChange": "Resource",
+          "ChangedApiName": "identityProvider",
+          "ChangeType": "Change",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:7472]:
```suggestion
          "ChangeType": "Deprecation",
```

**Review by FaithOmbongi — COMMENTED** [review]
**Deprecation** is now an accepted value for ChangeType.

File: api-reference/v1.0/resources/builtinidentityprovider.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+|:---------------|:--------|:----------|
+|[List](../api/identityproviderbase-list.md)|[identityProviderBase](../resources/identityproviderbase.md) collection|Retrieve all identity providers configured in a tenant including the built-in identity providers.|
+|[Get](../api/identityproviderbase-get.md) |builtInIdentityProvider|Retrieve properties of an built-in identity provider.|
+|[List available provider types](../api/identityproviderbase-list-availableprovidertypes.md)|String collection|Retrieve all available identity provider types available in the tenant.|
+
+## Properties
+
+|Property|Type|Description|
+|:---------------|:--------|:----------|
+|id|String|The identifier of the identity provider. Inherited from [identityProviderBase](../resources/identityproviderbase.md). Read-only.|
```
[the remark this answers is not in the document]
Comment (namkedia) [reply api-reference/v1.0/resources/builtinidentityprovider.md:31]: This comes from the identityproviderbase

File: api-reference/v1.0/resources/builtinidentityprovider.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+<!-- {
+  "blockType": "resource",
+  "@odata.type": "microsoft.graph.builtInIdentityProvider"
+} -->
+
+```json
+{
+    "id": "String",
+    "identityProviderType": "String",
+    "displayName": "String",
```
[the remark this answers is not in the document]
Comment (namkedia) [reply api-reference/v1.0/resources/builtinidentityprovider.md:48]: This comes from the identityproviderbase

File: api-reference/v1.0/resources/openidconnectidentityprovider.md [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+|domainHint|String|The domain hint can be used to skip directly to the sign-in page of the specified identity provider, instead of having the user make a selection among the list of available identity providers.|
+|metadataUrl|String|The URL for the metadata document of the OpenID Connect identity provider. Every OpenID Connect identity provider describes a metadata document that contains most of the information required to perform sign-in. This includes information such as the URLs to use and the location of the service's public signing keys. The OpenID Connect metadata document is always located at an endpoint that ends in `.well-known/openid-configuration`. Provide the metadata URL for the OpenID Connect identity provider you add. Read-only. Required.|
+|responseMode|String|The response mode defines the method used to send data back from the custom identity provider to Azure AD B2C. Possible values: `form_post`, `query`. Required.|
+|responseType|String|The response type describes the type of information sent back in the initial call to the authorization_endpoint of the custom identity provider. Possible values: `code` , `id_token` , `token`.  Required.|
```
[the remark this answers is not in the document]
Comment (namkedia) [reply api-reference/v1.0/resources/openidconnectidentityprovider.md]: remove oidc from v1.0

File: api-reference/v1.0/resources/claimsmapping.md [hunk 19]
```diff
@@ -0,0 +1,45 @@
+---
+title: "claimsMapping resource type"
```
[the remark this answers is not in the document]
Comment (namkedia) [reply api-reference/v1.0/resources/claimsmapping.md:2]: remove oidc from v1.0 doc

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Proposed minor changes. Just a little clean up.

**Review by FaithOmbongi — APPROVED** [review]

File: api-reference/v1.0/api/identityproviderbase-delete.md [hunk 20]
```diff
@@ -0,0 +1,80 @@
+---
+title: "Delete identityProvider"
+description: "Delete an identityProvider."
+localization_priority: Normal
+doc_type: apiPageType
+author: "namkedia"
+ms.prod: "identity-and-sign-in"
+---
+
+# Delete identityProvider
```
Comment (Lauragra) [inline api-reference/v1.0/api/identityproviderbase-delete.md:10]: Should the topic title here be "Delete socialIdentityProvider", and the file name be socialidentityprovider-delete? It seems that the file name, title, and description do not match.

File: api-reference/v1.0/toc.yml [hunk 21]
```diff
@@ -4684,17 +4684,38 @@ items:
             - name: Dismiss
               href: api/riskyuser-dismiss.md
         - name: Identity provider
-          href: resources/identityprovider.md
           items:
-          - name: List identity providers
+          - name: Identity provider base
+            href: resources/identityproviderbase.md
+          - name: Built-in identity provider
```
Comment (Lauragra) [inline api-reference/v1.0/toc.yml:4690]: The link to this v1.0 resource topic does not appear to be working on the review site; please double check the TOC entries.

Comment (Lauragra) [inline api-reference/v1.0/toc.yml:4690]: Can you help me understand the reason for this TOC structure? From the TOC, is it clear to users which of the operations apply to which resources? It seems that different methods apply to the built-in and social identity providers based on the Methods tables, but they are grouped together in the TOC.

File: api-reference/v1.0/api/identityproviderbase-delete.md (hunk 20 above)
Comment (namkedia) [reply api-reference/v1.0/api/identityproviderbase-delete.md:10]: No, we need to add other identityprovider types in the future. it should be delete identity provider.

File: api-reference/v1.0/toc.yml (hunk 21 above)
Comment (namkedia) [reply api-reference/v1.0/toc.yml:4690]: We have grouped all the identity provider types in the same set called identity provider and not separated it. Two more types will be added to this and our hope is to have them all represented in identity providers.

**Pull request merged**